pkg/controller/pod.go

```diff
@@ -202,15 +214,93 @@ func (c *Controller) syncPod(ctx context.Context, pod *corev1.Pod) error {
 		}
 	}
 
-	klog.V(4).Infof("Registering instance %s under service %+v", pod.Name, cloudmapConfig)
+	cloudMapServiceKey := *(awssdk.String(*cloudmapConfig.NamespaceName)) + "-" +
+		*(awssdk.String(*cloudmapConfig.ServiceName))
+
+	existingItem, exists, _ := c.cloudMapInstanceCache.Get(&CloudMapInstanceCacheItem{
+		key: cloudMapServiceKey,
+	})
+
+	var serviceInstanceSummary map[string]bool
+
+	if exists {
+		if healthy, ok := existingItem.(*CloudMapInstanceCacheItem).InstanceSummary[pod.Status.PodIP]; ok {
+			//Right now we're not logging pod/instance health info to CloudMap. So, if it is in cache then it is
+			//considered healthy. *TODO - v1.0* */
+			klog.V(4).Info("Pod Instance already registered. Nothing to do for this pod; Healthy?: ", healthy)
+			return nil
+		}
+		serviceInstanceSummary = existingItem.(*CloudMapInstanceCacheItem).InstanceSummary
+		if serviceInstanceSummary == nil { serviceInstanceSummary = make(map[string]bool) }
+		serviceInstanceSummary[pod.Status.PodIP] = true
+	} else {
+		//Retrieve CloudMap instances for this service
+		serviceInstanceSummary := make(map[string]bool)
+		registered := false
+
+		appmeshCloudMapConfig := &appmesh.AwsCloudMapServiceDiscovery{
+			NamespaceName: awssdk.String(*cloudmapConfig.NamespaceName),
```

Comment:
it's not needed. `awssdk.String(*cloudmapConfig.NamespaceName)` == `cloudmapConfig.NamespaceName`

Reply:
Yeah, was just following the notation followed in other files. Can change it